### ISSUE TITLE
No agent / no service on OS X 10.11.3

### ISSUE THREAD

**Issue by reckart** [pydio/pydio-sync#118]
Since a while, I see only grey "no agent" and "no service" items in the pydio sync UI on OS X.

* looking at the running processes, I see two `pydio-agent`processes are there under my user.
* looking at the output of lsof, I see that one of the processes writes its log to `/Data/pydio.log` instead of the usual place under `~/Library` (the place that "open logs" in the about dialog points me to), but the log only says this (repeatedly)

```
2016-01-20 18:26:06 INFO    140735234678784 MainThread logging:1627 | info | -----------------------------------------------
2016-01-20 18:26:06 INFO    140735234678784 MainThread logging:1627 | info | Starting agent locally on http://localhost:5556/
2016-01-20 18:26:06 INFO    140735234678784 MainThread logging:1627 | info | ------------------------------------------------
```

* if I kill the `pydio-agent` processes run `/Applications/PydioSync.app/Contents/Resources/pydio-agent` manually on a terminal, all is fine (until I close the terminal of course)

I am using the OS X pydio client PydioSync v1.0.3 - December 2 2015

**Comment by 7omate** [comment]
Please open a topic on our forum, https://pydio.com/forum/f/forum/troubleshooting/pydiosync/
We'll release a new version soon. If you can pinpoint any changes you made that might have caused this behavior that'd be great.

**Comment by reckart** [comment]
I posted here because I already have a github account. If I had to post on the forum, I wouldn't have done so in the first place because I don't care very much to register there. I had a look at the forum before I posted here and tried to locate suitable solutions that worked for others, but didn't find anything - except for important pointers to log files or running the agent from the command line.

I'm afraid, I cannot pinpoint any exact changes. The only thing I can say is that the last entry in the log files under `~/Library/...` are from 2015-12-12 - so the problem might coincide with an update to the PydioSync 1.0.3 or maybe with an OS upgrade.

Feel free to close this thread if you do not care following up on problems via this issue tracker.

**Comment by 7omate** [comment]
Reminds me of this:
https://pydio.com/forum/f/topic/pydiosync-v1-0-3-on-macosx-10-11-1-issue/
What happens when you run the agent manually? /Applications/PydioSync.app/Contents/Resources/pydio-agent

**Comment by reckart** [comment]
I found the forum post you cite as well, but when I run the agent manually, I do not see the kind of error reported in the forum post. Running the agent manually makes the menu items in the PydioSync UI available (no longer grey) and the agent runs and synchronizes normally:

```
...
15:09:44 INFO    140735234678784 MainThread -----------------------------------------------
15:09:44 INFO    140735234678784 MainThread Starting agent locally on http://localhost:5556/
15:09:44 INFO    140735234678784 MainThread ------------------------------------------------
15:09:44 INFO    140735234678784 MainThread Structure and Integrity of SQLite file /Users/.../Library/Application Support/Pydio/.../changes.sqlite is intact
15:09:44 INFO    140735234678784 MainThread Structure and Integrity of SQLite file /Users/.../Library/Application Support/Pydio/.../pydio.sqlite is intact
15:09:44 INFO    123145310715904 Thread-1 Scanning for changes since last application launch
15:09:44 INFO    140735234678784 MainThread Structure and Integrity of SQLite file /Users/.../Library/Application Support/Pydio/.../changes.sqlite is intact
15:09:44 INFO    140735234678784 MainThread Structure and Integrity of SQLite file /Users/.../Library/Application Support/Pydio/.../pydio.sqlite is intact
15:09:44 INFO    123145314922496 Thread-3 Scanning for changes since last application launch
15:09:44 INFO    123145319129088 Thread-4 Starting permanent monitor
15:09:44 INFO    123145314922496 Thread-3 Loading remote changes with sequence 1414
15:09:44 INFO    123145332363264 Thread-2 Starting permanent monitor
15:09:44 INFO    123145310715904 Thread-1 Loading remote changes with sequence 1414
...
```

Do it appears that just when the agent is started via PydioSync UI, that it is somehow using the wrong folder for logging (`/Data/`) and that the agent processes are hanging.

I had a quick look around how to produce a stack trace for running python processes, but didn't find anything easy on OS X. If you have an idea how I could (easily) provide you with a stack trace, it might give a clue where and why the agents are hanging.

**Comment by 7omate** [comment]
Maybe try to run /Applications/PydioSync.app/Contents/MacOS/pydio-ui
Do you have any clue what created /Data on your Mac?
The only alternatives are to update to 1.0.4 or reinstall... Quit and restart PydioSync

**Comment by reckart** [comment]
I'm still facing the problem, but didn't find the time yet to dig deeper into it. Right now, I'm helping myself with starting the agent on the command line from time to time. I have it on my todo list and get back to you with input about running pydio-ui as soon as I can.

**Comment by 7omate** [comment]
Ok please keep me posted.

**Comment by reckart** [comment]
Ok, sorry for the delay. I have tried running the pydio-ui from the command line and this is what it produces:

```
/Applications/PydioSync.app/Contents/MacOS/pydio-ui
Starting agent via launchctl command.
QSslSocket: cannot resolve TLSv1_1_client_method
QSslSocket: cannot resolve TLSv1_2_client_method
QSslSocket: cannot resolve TLSv1_1_server_method
QSslSocket: cannot resolve TLSv1_2_server_method
QSslSocket: cannot resolve SSL_select_next_proto
QSslSocket: cannot resolve SSL_CTX_set_next_proto_select_cb
QSslSocket: cannot resolve SSL_get0_next_proto_negotiated
  -- HTTPMANAGER --   :     "Server set to http://127.0.0.1:5556"
  -- HTTPMANAGER --   :     "Username set to XXXXXXXXXXXXXX"
  -- HTTPMANAGER --   :     "Providing authentication with name XXXXXXXXXXXXXX"
  -- HTTPMANAGER --   :     "Providing authentication with name XXXXXXXXXXXXXX"
  -- HTTPMANAGER --   :     "Providing authentication with name XXXXXXXXXXXXXX"
  -- HTTPMANAGER --   :     "Providing authentication with name XXXXXXXXXXXXXX"
  -- HTTPMANAGER --   :     "Providing authentication with name XXXXXXXXXXXXXX"
ERROR WITH UPDATE REQUEST
  -- HTTPMANAGER --   :     "Skipping authentication with name XXXXXXXXXXXXXX"
  -- HTTPMANAGER --   :     "-----------------------MAX CONNECTION ATTEMPTS REACHED, CLEARING JOBS-----------------------"
UPDATING PORTS FROM CONFIG FILE
  -- HTTPMANAGER --   :     "Server set to http://127.0.0.1:5556"
  -- HTTPMANAGER --   :     "Username set to XXXXXXXXXXXXXX"
  -- HTTPMANAGER --   :     "Providing authentication with name XXXXXXXXXXXXXX"
  -- HTTPMANAGER --   :     "Providing authentication with name XXXXXXXXXXXXXX"
  -- HTTPMANAGER --   :     "Providing authentication with name XXXXXXXXXXXXXX"
  -- HTTPMANAGER --   :     "Providing authentication with name XXXXXXXXXXXXXX"
  -- HTTPMANAGER --   :     "Providing authentication with name XXXXXXXXXXXXXX"
  -- HTTPMANAGER --   :     "Skipping authentication with name XXXXXXXXXXXXXX"
  -- HTTPMANAGER --   :     "-----------------------MAX CONNECTION ATTEMPTS REACHED, CLEARING JOBS-----------------------"
UPDATING PORTS FROM CONFIG FILE
  -- HTTPMANAGER --   :     "Server set to http://127.0.0.1:5556"
  -- HTTPMANAGER --   :     "Username set to XXXXXXXXXXXXXX"
  -- HTTPMANAGER --   :     "Providing authentication with name XXXXXXXXXXXXXX"
  -- HTTPMANAGER --   :     "Providing authentication with name XXXXXXXXXXXXXX"
  -- HTTPMANAGER --   :     "Providing authentication with name XXXXXXXXXXXXXX"
  -- HTTPMANAGER --   :     "Providing authentication with name XXXXXXXXXXXXXX"
  -- HTTPMANAGER --   :     "Providing authentication with name XXXXXXXXXXXXXX"
  -- HTTPMANAGER --   :     "Skipping authentication with name XXXXXXXXXXXXXX"
  -- HTTPMANAGER --   :     "-----------------------MAX CONNECTION ATTEMPTS REACHED, CLEARING JOBS-----------------------"
```

This goes on forever. The XXXXX is actually something that looks like a random string.

**Comment by reckart** [comment]
I forgot to say: the items in the menu are also greyed out when I start pydio-ui from the command line.
After stopping the processes (CTRL-C), the agent processes are still around.

If I then try to restart pydio-ui from the command line, I get an extra error line:

```
Starting agent via launchctl command.
QSslSocket: cannot resolve TLSv1_1_client_method
QSslSocket: cannot resolve TLSv1_2_client_method
QSslSocket: cannot resolve TLSv1_1_server_method
QSslSocket: cannot resolve TLSv1_2_server_method
QSslSocket: cannot resolve SSL_select_next_proto
QSslSocket: cannot resolve SSL_CTX_set_next_proto_select_cb
QSslSocket: cannot resolve SSL_get0_next_proto_negotiated
  -- HTTPMANAGER --   :     "Server set to http://127.0.0.1:5556"
  -- HTTPMANAGER --   :     "Username set to XXXXXXXXXXXXXX"
  -- HTTPMANAGER --   :     "Providing authentication with name XXXXXXXXXXXXXX"
  -- HTTPMANAGER --   :     "Providing authentication with name XXXXXXXXXXXXXX"
  -- HTTPMANAGER --   :     "Providing authentication with name XXXXXXXXXXXXXX"
  -- HTTPMANAGER --   :     "Providing authentication with name XXXXXXXXXXXXXX"
  -- HTTPMANAGER --   :     "Providing authentication with name XXXXXXXXXXXXXX"
  -- HTTPMANAGER --   :     "Skipping authentication with name XXXXXXXXXXXXXX"
  -- HTTPMANAGER --   :     "-----------------------MAX CONNECTION ATTEMPTS REACHED, CLEARING JOBS-----------------------"
UPDATING PORTS FROM CONFIG FILE
  -- HTTPMANAGER --   :     "Server set to http://127.0.0.1:5556"
  -- HTTPMANAGER --   :     "Username set to XXXXXXXXXXXXXX"
ERROR WITH UPDATE REQUEST
```

and then the same endless sequence as outlined in the previous comment.

**Comment by reckart** [comment]
I just upgraded to 1.0.4 but the erratic behavior and the logs are essentially the same. I might not that I am running a virus scanner and a private firewall - but I have always used them and PydioSync worked nicely. Also the behavior does not change if I disable the virus scanner and firewall.

**Comment by 7omate** [comment]
When exactly do you have issues? When you turn on your computer? When you quit (wait ~10s) and launch PydioSync from applications? 
What happens when you [code] open http://127.0.0.1:5556 [/code] also check the content of /Library/Application Support/Pydio/ports_config it needs to match the info displayed in pydio.log

**Comment by reckart** [comment]
Problem happens when starting the computer new (with PydioSync auto-starting on login) as well as when running it manually from the "Applications" folder.

I removed the strange "Data" folder in my HDD root now that PydioSync appears to be using by default - at least, now PydioSync seems to be using the settings and logs in "~/Library/Application Support/Pydio" again.

If I start pydio-ui from the command line now, I get a working UI menu again.

I also tried running PydioSync from Applications now - but the icon quickly disappeared again from the menu bar. 

After that, I killed all the pydio processes, tried starting pydio-ui from the command line: ok

Again, I killed all the pydio processes, tried starting PydioSync from Application: also ok now

So the solution seems to have been: delete the unexplainable "/Data" folder and give PydioSync a fresh start. I still have no idea where the "/Data" folder came from. If it reappears, I'll drop a note.
